src/client/ui/properties/VideoNodeEditor.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component, Fragment } from "react";
+import PropTypes from "prop-types";
+import NodeEditor from "./NodeEditor";
+import InputGroup from "../inputs/InputGroup";
+import BooleanInput from "../inputs/BooleanInput";
+import SelectInput from "../inputs/SelectInput";
+import NumericInput from "../inputs/NumericInput";
+import CompoundNumericInput from "../inputs/CompoundNumericInput";
+import { AudioType, DistanceModelType } from "../../editor/objects/Video";
+
+const audioTypeOptions = Object.values(AudioType).map(v => ({ label: v, value: v }));
+
+const distanceModelOptions = Object.values(DistanceModelType).map(v => ({ label: v, value: v }));
+
+export default class VideoNodeEditor extends Component {
+  static propTypes = {
+    editor: PropTypes.object,
+    node: PropTypes.object
+  };
+
+  static iconClassName = "fa-video";
+
+  setProp(propName) {
```

Comment:
Very minor but the extra binding here isn't needed, and the name feels a bit misleading since this actually doesn't set a prop, it returns a function that does. I probably would have wrote this as a function outside the class like:
```
function propSetter(propName) {
  return function(value) {
    this.props.editor.setNodeProperty(this.props.node, propName, value);
  }
}
```

and used:
```
onChangeControls: propSetter("controls"),
```

Reply:
`this` will not be bound to the correct value. I think renaming it to `createPropSetter` is a good idea.

Maybe this is the cleanest implementation?
```
constructor(props) {
    super(props);
    const createPropSetter = propName => value => this.props.editor.setNodeProperty(this.props.node, propName, value);
    this.onChangeControls = createPropSetter("controls");
    this.onChangeAutoPlay = createPropSetter("autoPlay");
    this.onChangeLoop = createPropSetter("loop");
    this.onChangeAudioType = createPropSetter("audioType");
    this.onChangeVolume = createPropSetter("volume");
    this.onChangeDistanceModel = createPropSetter("distanceModel");
    this.onChangeRolloffFactor = createPropSetter("rolloffFactor");
    this.onChangeRefDistance = createPropSetter("refDistance");
    this.onChangeMaxDistance = createPropSetter("maxDistance");
    this.onChangeConeInnerAngle = createPropSetter("coneInnerAngle");
    this.onChangeConeOuterAngle = createPropSetter("coneOuterAngle");
    this.onChangeConeOuterGain = createPropSetter("coneOuterGain");
  }
```